### ISSUE TITLE
Upload error on devices with locale whose numeric character is non-ascii

### ISSUE THREAD

**Issue by kfujino7** [dropbox/dropbox-sdk-java#119]
Android users with Arabic setting are seeing following error on uploading the file.

> com.dropbox.core.BadRequestException: Error in call to API function "files/upload": HTTP header "Dropbox-API-Arg": client_modified: ascii

Uploading the file with standard way,,,
```
FileMetadata metadata = dbxClient.files().uploadBuilder(path)
                    .withClientModified(new Date(localFile.lastModified()))
                    .uploadAndFinish(inputStream);
```

The cause seems `com.dropbox.core.stone.Util` `formatTimestamp` return non-ascii character depending on the device's locale setting.

```
public static String formatTimestamp(Date timestamp) {
        DateFormat format = new SimpleDateFormat(DATE_TIME_FORMAT);
        format.setCalendar(new GregorianCalendar(UTC));
        return format.format(timestamp);
    }
```
	
For example, with the arabic configured device (Locale ar_EG) , it returns like
 ٢٠١٧-٠٦-٢٠T١٢:٣٧:٤٦Z 
 for 
 2017-06-20T12:37:46Z
 
and DropBox server doesn't like it. 

We may want to specify the locale like Locale.English for SimpleDateFormat?
Tested on 2.1.2 but should be same result for the latest 3.0.3.

Thank you.

**Comment by greg-db** [comment]
Thanks! We'll look into it.

**Comment by bazimogmbh** [comment]
I had the same issue, my workaround was not to set cleintmodified date in uploadbuilder, but i needed that info, so i fetched it after the upload was finished from uploadresult.

**Comment by ControlAdad** [comment]
I have same problem, can't use *withClientModified*, I receive this error :
Error in call to API function "files/upload": HTTP header "Dropbox-API-Arg": client_modified: ascii

My Locale is Persian

**Comment by greg-db** [comment]
This is fixed as of [v3.0.6](https://github.com/dropbox/dropbox-sdk-java/releases/tag/v3.0.6).
